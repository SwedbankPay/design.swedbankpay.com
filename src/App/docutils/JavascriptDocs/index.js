import React from "react";
import { Link } from "react-router-dom";

import CodeTags from "@components/CodeTags";

import DocElement from "@docutils/JavascriptDocElement";

const OpenDocs = ({ componentName }) => (
    <>
        <DocElement
            code={<CodeTags type="secondary" code={`dg.${componentName}.open(<${componentName}-id>)`} params={[`<${componentName}-id>`]} />}
            description={
                <>
                    Opens the given {componentName.toLowerCase()}
                </>
            }
        />
    </>
);

const CloseDocs = ({ componentName }) => (
    <>
        <DocElement
            code={<CodeTags type="secondary" code={`dg.${componentName}.close(<${componentName}-id>)`} params={[`<${componentName}-id>`]} />}
            description={
                <>
                    Closes the given {componentName}
                </>
            }
        />
    </>
);

const JavascriptDocs = ({ componentName, open, close, others }) => (
    <>
        <ul className="item-list">
            <DocElement
                code={<CodeTags type="secondary" code={`dg.${componentName}.init()`} />}
                description={
                    <>
                        <Link to="/docs/getting-started/javascript">Initializes</Link> {componentName}
                    </>
                }
            />
            {open ? <OpenDocs componentName={componentName} /> : null}
            {close ? <CloseDocs componentName={componentName} /> : null}
            {others && others.map((DocComponent, i) => <DocComponent key={i} componentName={componentName} />)}
        </ul>
    </>
);

export default JavascriptDocs;

/* For testing */
export { OpenDocs, CloseDocs };
