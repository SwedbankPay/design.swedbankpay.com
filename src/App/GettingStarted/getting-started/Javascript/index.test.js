import React from "react";
import { shallow } from "enzyme";

import JavascriptDocs, { AboutInit } from "./index";

describe("GettingStarted: JavascriptDocs", () => {
    it("is defined", () => {
        expect(JavascriptDocs).toBeDefined();
    });

    it("renders", () => {
        const wrapper = shallow(<JavascriptDocs />);

        expect(wrapper).toMatchSnapshot();
    });

    describe("AboutInit", () => {
        it("is defined", () => {
            expect(AboutInit).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<AboutInit />);

            expect(wrapper).toMatchSnapshot();
        });
    });
});

