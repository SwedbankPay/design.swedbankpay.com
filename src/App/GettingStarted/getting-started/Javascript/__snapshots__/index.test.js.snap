// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GettingStarted: JavascriptDocs AboutInit renders 1`] = `
<Fragment>
  <h2
    id="about-init"
  >
    When to initialize
  </h2>
  <p>
    Our script runs 
    <DgScript
      component="script"
      func="initAll"
    />
     on DOMContentLoaded making it so you have to run 
    <DgScript
      component="<component_name>"
      func="init"
    />
     manually when a component is rendered after the event has fired.
  </p>
  <h3>
    Manual initializing
  </h3>
  <p>
    To initialize components use 
    <DgScript
      component="<component_name>"
      func="init"
    />
    . If the function is called without arguments it will target all components with class 
    <CodeTags
      code="<component_name>"
      type="secondary"
    />
     on the page. Initializing only one component can be done by passing an ID, it must match an existing components ID and will print a warning message if the value is invalid.
  </p>
  <p>
    If only one component is initialized then the init function will return one object. If several components are initialized it will return an array of objects.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="javascript"
  >
    dg.component.init();

    dg.component.init("demo-component-id");

  </ComponentPreview>
  <Alert
    type="warning"
  >
    Additional functions is available for some components. Read about them on the components documentation page.
  </Alert>
</Fragment>
`;

exports[`GettingStarted: JavascriptDocs renders 1`] = `
<DocContainer
  docToc={true}
>
  <p
    className="lead"
  >
    How to utilize our 
    <DgScript
      component="<component_name>"
      func="init"
    />
     function.
  </p>
  <AboutInit />
</DocContainer>
`;
